openedx/features/course_duration_limits/access.py

```diff
@@ -130,19 +131,41 @@ def register_course_expired_message(request, course):
             )
         )
     else:
-        upgrade_message = _('Your access to this course expires on {expiration_date}. \
-                    {a_open}Upgrade now {sronly_span_open}to retain access past {expiration_date}.\
-                    {span_close}{a_close}{sighted_only_span_open}for unlimited access.{span_close}')
+        enrollment = CourseEnrollment.get_enrollment(request.user, course.id)
+        if enrollment is None:
+            return
+
+        upgrade_deadline = enrollment.upgrade_deadline
+        if upgrade_deadline is None:
+            return
+        now = timezone.now()
+        if now < upgrade_deadline:
```

Comment:
Testing this on stage I see the "course_upgrade_deadline" displayed instead of the "upgrade_deadline" even though we aren't passed the upgrade_deadline.

I think this needs to be:
```python
if now > upgrade_deadline:
    upgrade_deadline = enrollment.course_upgrade_deadline
```